sys/include/net/sock/dtls.h

```diff
@@ -0,0 +1,603 @@
+ * if (!ipv6_addr_from_str((ipv6_addr_t *)remote.addr.ipv6, addr_str)) {
+ *     puts("Error parsing destination address");
+ *     return;
+ * }
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * After the UDP sock is created, we can proceed with creating the DTLS sock.
+ * Before sending the packet, we must first establish a session with the remote
+ * endpoint using @ref sock_dtls_establish_session(). If the handshake is
+ * successfull and the session is established, we can use its `remote` to send
+ * packets to it with @ref sock_dtls_send().
+ *
+ * If an error occurs during any of these operation, the session is terminated
+ * and the sock is closed using @ref sock_dtls_terminate_session() and
+ * @ref sock_dtls_close().
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+ * res = sock_dtls_create(&dtls_sock, &udp_sock, DTLS_SOCK_CLIENT_TAG, 0);
+ * if (res < 0) {
+ *     puts("Error creating DTLS sock");
+ *     return;
+ * }
+ *
+ * res = sock_dtls_establish_session(&dtls_sock, &remote, &session);
+ * if (res < 0) {
+ *     printf("Error establishing session: %d\n", (int)res);
+ *     goto end;
+ * }
+ *
+ * res = sock_dtls_send(&dtls_sock, &session, data, datalen);
+ * if (res < 0) {
+ *     printf("Error sending DTLS message: %d\n", (int)res);
+ *     goto end;
+ * }
+ *
+ * res = sock_dtls_recv(&dtls_sock, &session, rcv, sizeof(rcv), SOCK_NO_TIMEOUT);
+ * if (res < 0) {
+ *     printf("Error receiving DTLS message: %d\n", (int)res);
+ *     goto end;
+ * }
+ *
+ * end:
+ * puts("Terminating");
+ * sock_dtls_terminate_session(&dtls_sock, &session);
+ * sock_dtls_close(&dtls_sock);
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * @{
+ *
+ * @file
+ * @brief   DTLS sock definitions
+ *
+ * @author  Aiman Ismail <muhammadaimanbin.ismail@haw-hamburg.de>
+ * @author  Martine Lenders <m.lenders@fu-berlin.de>
+ * @author  Raul A. Fuentes Samaniego <raul.fuentes-samaniego@inria.fr>
+ * @author  Daniele Lacamera <daniele@wolfssl.com>
+ * @author  Ken Bannister <kb2ma@runbox.com>
+ */
+
+#ifndef NET_SOCK_DTLS_H
+#define NET_SOCK_DTLS_H
+
+#include <assert.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <sys/types.h>
+
+#include "net/sock.h"
+#include "net/sock/udp.h"
+#include "net/credman.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Protocol version number
+ */
+typedef struct {
+    uint8_t major;  /**< Major version number */
+    uint8_t minor;  /**< Minor version number */
+} ProtocolVersion;
+
+#define DTLSv1      {254, 255}  /**< DTLS 1.0 version RFC4347#Section-4.1 */
+#define DTLSv1_2    {254, 253}  /**< DTLS 1.2 version RFC6347#Section-4.1 */
+
+#define DTLS_CLIENT (1),    /**< Endpoint client role */
+#define DTLS_SERVER (2),    /**< Endpoing server role */
+
+/**
+ * @brief Method of connecting to remote
+ */
+typedef struct {
+    ProtocolVersion version;    /**< DTLS version number */
+    uint8_t role;               /**< Role of the endpoint */
```

Comment:
You can then link it here:

```suggestion
    /**
     * @brief Role of the endpoint 
     * @see [DTLS roles](@ref sock_dtls_roles))
     */
    uint8_t role;
```

Reply:
Done.